kernel/src/insn_decode/decode.rs

```diff
@@ -523,6 +583,32 @@ fn ioio_perm<I: InsnMachineCtx>(mctx: &I, port: u16, size: Bytes, io_read: bool)
     }
 }
 
+#[inline]
+fn read_bytereg<I: InsnMachineCtx>(mctx: &I, reg: Register, lhbr: bool) -> u8 {
+    let data = mctx.read_reg(reg);
```

Comment:
I guess you can cast to `u8` at the beginning to avoid cast duplicate: 
```suggestion
    let data = mctx.read_reg(reg) as u8;
```

Reply:
We may need to get the high-byte of data if the lhbr is true in the below if case. If cast data as u8 here, the high-byte is lost. How about change below code as `(if lhbr { data >> 8 } else { data }) as u8`?

Reply:
Yes, this is better indeed :)

Reply:
I will update with this change. Thanks!